### DISCUSSION TITLE
Understanding the code that produces the number of built objects in the log

### DISCUSSION THREAD

**Discussion by sadeka4** [dbt-labs/dbt-core#11172] [Q&A] [answered]
I am trying to understand the code that produces the number of built objects in the log. An example looks like this.

`03:30:07  46 of 59 OK created sql table model my_dataset.fct_generation`

I'd like to know how DBT generates a total of 59 objects in the above line. Which resource types are considered to be included while counting this total and which objects are actually run by DBT afterwards.

Any help would be greatly appreciated!
Thank you.

**Comment by QMalcolm** [comment, marked as answer]
Hi @sadeka4 so the log in question is an instance of the [LogModelResult](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/events/types.py#L1365) event. There are additional similar events in dbt-core: [LogFreshnessResult](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/events/types.py#L1439), [LogNodeNoOpResult](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/events/types.py#L1481), [LogSnapshotResult](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/events/types.py#L1393), [LogSeedResult](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/events/types.py#L1415), and [LogTestResult](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/events/types.py#L1301). Something similar about all of these is that they all have a `total` attribute. This `total` is the number of nodes selected for execution in the given run, which is calculated in [_runtime_initialize](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/task/runnable.py#L194) of `GraphRunnableTask`. We can see in this function that the total `num_nodes` is the "selected for execution" nodes that are one of
* Exposure
* UnitTest
* SavedQuery
* Source
* "nodes" (Can be any "[ManifestNode](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/contracts/graph/nodes.py#L1738)")
  * Analysis
  * SignularTestNode
  * HookNode
  * ModelNode
  * SqlNode
  * GenericTestNode
  * SnapshotNode
  * UnitTestNode
  * SeedNode

There is one event that is slight different, [LogBatchResult](https://github.com/dbt-labs/dbt-core/blob/57e279cc1beb95356cfc7c84b86c890fb9e3e7de/core/dbt/events/types.py#L1748). The `total` number for these log events is the number of batches being executed for the given microbatch model. Its log line will also look slightly different, something like `03:30:07 BATCH 1 of 7 OK created batch...`

**Reply by sadeka4** [reply]
Many thanks, @QMalcolm for clarifying this. Much appreciated!
